core/conf/config.go

```diff
@@ -0,0 +1,138 @@
+package conf
+
+import (
+	"errors"
+	"net/url"
+	"os"
+	"strings"
+
+	thrown "github.com/0chain/errors"
+	"github.com/spf13/viper"
+)
+
+// Config settings from ~/.zcn/config.yaml
+// block_worker: http://198.18.0.98:9091
+// signature_scheme: bls0chain
+// min_submit: 50
+// min_confirmation: 50
+// confirmation_chain_length: 3
+// max_txn_query: 5
+// query_sleep_time: 5
+// # # OPTIONAL - Uncomment to use/ Add more if you want
+// # preferred_blobbers:
+// #   - http://one.devnet-0chain.net:31051
+// #   - http://one.devnet-0chain.net:31052
+// #   - http://one.devnet-0chain.net:31053
+type Config struct {
+	// BlockWorker the url of 0dns's network api
+	BlockWorker string
+	// PreferredBlobbers preferred blobbers on new allocation
+	PreferredBlobbers []string
+
+	// MinSubmit mininal submit from blobber
+	MinSubmit int
+	// MinConfirmation mininal confirmation from sharders
+	MinConfirmation int
+	// CconfirmationChainLength minial confirmation chain length
+	ConfirmationChainLength int
+
+	// additional settings depending network latency
+	// MaxTxnQuery maximum transcation query from sharders
+	MaxTxnQuery int
+	// QuerySleepTime sleep time before transcation query
+	QuerySleepTime int
+
+	// SignatureScheme signature scheme
+	SignatureScheme string
+	// ChainID which blockchain it is working
+	ChainID string
+}
+
+// LoadConfigFile load and parse Config from file
+func LoadConfigFile(file string) (Config, error) {
```

Comment:
You do not seem to call LoadConfig from inside `gosdk`. Is the intention that users of `gosdk` should call LoadConfig before each batch of `gosdk` method call?
````
   sdk.SetConfig(LoadConfi(".zcn"))
   sdk.DoSomethig(someParm)
```

This will require changes to `zbox`, `zwallet` etc.

Reply:
Yes. I have explain it 

> `talk is cheap, let's show it in code`
> 
> @Sriep @moldis I have refactored code again based on your reviews.
> 
> About `global variable` issue. I am prefer to do it in global variable too. It prevents to introduce break changes as less as possible. so I have reverted code to global variable. But `Load` is moved out from gosdk. zbox/zwallet has to load it by itself, and set global variable by `transaction.SetConfig`. eg
> 
> https://github.com/0chain/zwalletcli/blob/bac54c67d98defb4a4593c7ac2b44cfd7fd5494b/cmd/root.go#L84
> 
> https://github.com/0chain/zboxcli/blob/8f4a1801e47aeeae2bd7ace92c25599d85b62a58/cmd/root.go#L98
> 
> @Sriep please check [0chain/zwalletcli@bac54c6](https://github.com/0chain/zwalletcli/commit/bac54c67d98defb4a4593c7ac2b44cfd7fd5494b) and [0chain/zboxcli@8f4a180](https://github.com/0chain/zboxcli/commit/8f4a1801e47aeeae2bd7ace92c25599d85b62a58) . is it fine for you to review this PR now?
> 
> please help me to review this PR again @Sriep @moldis thanks.